Automattic/wp-calypso
# Revert "Update plan-specific flows to use generic plans step"

Reverts Automattic/wp-calypso#35047

Now that https://github.com/Automattic/wp-calypso/pull/35109 has been merged (remove index check in `removeUnneededSteps` we can attempt to revert the temporary fix for the broken transition to the ecommerce flow: https://github.com/Automattic/wp-calypso/issues/32506

## Testing

We wish to test that all of the plan-specific flows function as they used to. Pay particular attention to using the 'back' button and that it takes you to the correct step.

* From an incognito window, go to http://calypso.localhost:3000/start/personal
* Log in on the user step
* From the Site type step, select Online store, and it should take you to the domains step on the ecommerce-onboarding flow, and you should see: http://calypso.localhost:3000/start/ecommerce-onboarding/domains in the address bar
* Click back, and you should be on the site-type step
* Select any other site type (e.g. blog) and you should be on the site-topic-with-preview step (e.g. http://calypso.localhost:3000/start/personal/site-topic-with-preview )
* Continue on with the signup flow

Repeat the above steps for:

* http://calypso.localhost:3000/start/premium
* http://calypso.localhost:3000/start/business
* http://calypso.localhost:3000/start/free
* http://calypso.localhost:3000/start/

For each flow, ensure that a site can be created, or that you arrive on the correct checkout with the correct plan pre-selected.
